src/Provider/Netherlands/Netherlands.php

```diff
@@ -0,0 +1,99 @@
+        $holidays->add($this->getAscension($year, HolidayType::OFFICIAL | HolidayType::DAY_OFF));
+        if ($year >= 1891 && $year < 2014) {
+            $holidays->add($this->getQueensDay($year, HolidayType::OFFICIAL | HolidayType::DAY_OFF));
+        }
+        if ($year >= 2014) {
+            $holidays->add($this->getKingsDay($year, HolidayType::OFFICIAL | HolidayType::DAY_OFF));
+        }
+        if ($year >= 1946 && $liberationDay = $this->getLiberationDay($year, HolidayType::OFFICIAL)) {
+            $holidays->add($liberationDay);
+        }
+        $holidays->add($this->getWhitSunday($year, HolidayType::OFFICIAL | HolidayType::DAY_OFF));
+        $holidays->add($this->getWhitMonday($year, HolidayType::OFFICIAL | HolidayType::DAY_OFF));
+        $holidays->add($this->getChristmasDay($year, HolidayType::OFFICIAL | HolidayType::DAY_OFF));
+        $holidays->add($this->getSecondChristmasDay($year, HolidayType::OFFICIAL | HolidayType::DAY_OFF));
+
+        return $holidays;
+    }
+
+    private function getQueensDay(int $year, int $additionalType = HolidayType::OTHER): Holiday
+    {
+        $date = ($year < 1949) ? "08-31" : "04-30";
+        $date = \DateTime::createFromFormat(Holiday::CREATE_DATE_FORMAT, "{$year}-{$date}");
+        $weekDay = $date->format('w');
+        if ('0' === $weekDay) {
+            if($year < 1980) {
+                $date->add(new \DateInterval('P1D'));
+            } else {
+                $date->sub(new \DateInterval('P1D'));
+            }
+        }
+
+        return Holiday::create(HolidayName::QUEENS_DAY, $date->format(Holiday::DISPLAY_DATE_FORMAT), $additionalType | HolidayType::TRADITIONAL | HolidayType::GOVERNMENT_AGENCIES_CLOSED);
+    }
+
+    private function getKingsDay(int $year, int $additionalType = HolidayType::OTHER): Holiday
+    {
+        $date = \DateTime::createFromFormat(Holiday::CREATE_DATE_FORMAT, "{$year}-04-27");
+        $weekDay = $date->format('w');
+        if ('0' === $weekDay) {
+            $date->sub(new \DateInterval('P1D')); // Move to Saturday
+        }
+
+        return Holiday::create(HolidayName::KINGS_DAY, $date->format(Holiday::DISPLAY_DATE_FORMAT), $additionalType | HolidayType::TRADITIONAL | HolidayType::GOVERNMENT_AGENCIES_CLOSED);
+    }
+
+    private function getLiberationDay(int $year, int $additionalType = HolidayType::OTHER): ?Holiday
+    {
+        $date = \DateTime::createFromFormat(Holiday::CREATE_DATE_FORMAT, "{$year}-05-05");
+        $weekDay = $date->format('w');
+
+        if ($year < 1968) {
+            if ('0' === $weekDay) {
+                $date->add(new \DateInterval('P1D')); // Move to Monday
+            }
+        }
+
+        if ($year > 1968 && $year < 1990 && $year % 5 !== 0) {
+            return null;
+        }
+
+        return Holiday::create(HolidayName::DUTCH_LIBERATION_DAY, $date->format(Holiday::DISPLAY_DATE_FORMAT), $additionalType | HolidayType::TRADITIONAL | HolidayType::GOVERNMENT_AGENCIES_CLOSED);
```

Comment:
You're absolutely right! By just looking at the other countries, I wasn't sure why some of the HolidayTypes are in the helper methods and some are in the main `calculateHolidaysForYear` one, but I guess it doesn't matter very much in the end 😉

Reply:
I tried to add those types that are "inherent" to a holiday (such as if it is religious or traditional) directly in the helper methods and the "impact" (such as days off) in the caller, so that the helper methods can in theory be reused elsewhere. But this really matters only for common holidays in the helper traits, not in a concrete provider. So yes, every combination is fine and we could even simplify providers in the future.